digitalidentity/service.go

```diff
@@ -33,14 +36,47 @@ func CreateShareSession(httpClient requests.HttpClient, shareSession *ShareSessi
 		HTTPMethod: http.MethodPost,
 		BaseURL:    apiUrl,
 		Endpoint:   endpoint,
-		Headers:    nil,
+		Headers:    requests.AuthHeader(clientSdkId),
 		Body:       payload,
 	}.Request()
 	if err != nil {
 		return share, err
 	}
 
-	response, err := requests.Execute(httpClient, request, ShareURLHTTPErrorMessages, yotierror.DefaultHTTPErrorMessages)
+	response, err := requests.Execute(httpClient, request, ShareSessionHTTPErrorMessages, yotierror.DefaultHTTPErrorMessages)
+	if err != nil {
+		return share, err
+	}
+
+	defer response.Body.Close()
+
+	responseBytes, err := io.ReadAll(response.Body)
+	if err != nil {
+		return share, err
+	}
+
+	err = json.Unmarshal(responseBytes, &share)
+
+	return share, err
+}
+
+// GetSession get session info using the supplied sessionID
+func GetSession(httpClient requests.HttpClient, sessionID string, clientSdkId, apiUrl string, key *rsa.PrivateKey) (share ShareSession, err error) {
+	endpoint := identitySesssionCreationEndpoint
```

Comment:
wrong endpoint, should be `identitySessionRetrieval` (and should substitute the `%s`)

```suggestion
	endpoint := fmt.Sprintf(identitySessionRetrieval, sessionID)
```